### ISSUE TITLE
The rotation of somes elements are not applied

### ISSUE THREAD

**Issue by Sim3d** [xBimTeam/XbimGeometry#304]
Hi!

When I load a IFC file ([RotatedRoofElements.zip](https://github.com/xBimTeam/XbimGeometry/files/5850061/RotatedRoofElements.zip)) in the xbim Xplorer (built from source) two roof elements are not rotated as supposed to be. You can see the roof element below:

![RotatedRoof](https://user-images.githubusercontent.com/17932697/105366824-c535e900-5bcd-11eb-93fd-2c061c23e2d8.jpg)

Do you have an idea of what can cause the problem?

Thanks!

Simon

Note: 

1. The IFC file is attached.
2. The "xbim Xplorer" tool Assemblies are:

Xbim.Common	5.1.0.0	5.1.323.0
Xbim.Geometry.Engine.Interop	5.1.0.0	5.1.403.0
Xbim.Ifc	5.1.0.0	5.1.323.0
Xbim.Ifc2x3	5.1.0.0	5.1.323.0
Xbim.Ifc4	5.1.0.0	5.1.323.0
Xbim.IO.Esent	5.1.0.0	5.1.0.322
Xbim.IO.MemoryModel	5.1.0.0	5.1.323.0
Xbim.ModelGeometry.Scene	5.1.0.1	5.1.403.0
Xbim.Presentation	5.1.0.0	5.1.0.0
Xbim.Tessellator	5.1.0.0	5.1.323.0

**Comment by martin1cerny** [comment]
Confirmed this is a bug. Xbim seems not to be using `#79=IFCAXIS2PLACEMENT3D(#80,$,#81);`.
Here is the minimal file:

[roof_and_wall.zip](https://github.com/xBimTeam/XbimGeometry/files/5854577/roof_and_wall.zip)

![image](https://user-images.githubusercontent.com/773078/105469000-eb27c000-5c97-11eb-94eb-feb22ab888ca.png)

**Comment by Sim3d** [comment]
Thanks for the quick answer Martin!

When I look the definition of the IfcAxis2Placement3D, the RefDirection parameter is optional and the "Attribute definitions" documentation says:

If Axis and/or **RefDirection is omitted**, these directions are **taken from the geometric coordinate system**.

So in that case, you can probably fallback on the IfcDirection of the WorldCoordinateSystem (IfcAxis2Placement) given by the IfcGeometricRepresentationContext

#18 = IFCGEOMETRICREPRESENTATIONCONTEXT($, 'Model', 3, 1.E-5, #19, #23);
#19 = IFCAXIS2PLACEMENT3D(#20, #21, #22);
#21 = IFCDIRECTION((0., 0., 1.)); <====

https://standards.buildingsmart.org/IFC/RELEASE/IFC2x2/FINAL/HTML/ifcrepresentationresource/lexical/ifcgeometricrepresentationcontext.html

But maybe i'm wrong?

**Comment by martin1cerny** [comment]
There is a `WHERE` rule `AxisAndRefDirProvision` which says: 
> Either both, Axis and RefDirection are not given and therefore defaulted, or both shall be given.

And your file is breaking this rule. Therefore it would fail in IFC schema validation.

**Comment by Sim3d** [comment]
Oh! I see! The IFC documentation is tricky sometimes, sorry.

I tried to load the file into Solibri, Navisworks 2019 and SimpleBIM and all softwares displays the roof element correctly, a rotation is applied as expected...  So, would you have any idea of ​​a workaround to apply a rotation on the element like other BIM software? 

Thanks!

**Comment by martin1cerny** [comment]
Yes, this is a simple fallback. We will address it when we get to do some work on geometry engine.

**Comment by Sim3d** [comment]
Did you have the time to add the simple fallback?
